src/Utils.ps1

```diff
@@ -289,10 +290,24 @@ function Get-PromptPath {
     $pathInfo = $ExecutionContext.SessionState.Path.CurrentLocation
     $currentPath = if ($pathInfo.Drive) { $pathInfo.Path } else { $pathInfo.ProviderPath }
 
+    # Look up the git root
+    if ($abbrevGitDir) {
+        $gitPath = $(Get-GitDirectory)
+        # Up one level from `.git`
+        if ($gitPath) { $gitPath = $(Split-Path $gitPath -Parent) }
+    }
+
     $stringComparison = Get-PathStringComparison
 
+    # Abbreviate path by replacing beginning of path with - *iff* the path is under a git repository
+    if ($abbrevGitDir -and $currentPath -and $gitPath -and
+        $currentPath.StartsWith($gitPath, $stringComparison)) {
+        # Up another level to keep repo name in path
+        $removePath = $(Split-Path $gitPath -Parent)
+        $currentPath = "-" + $currentPath.SubString($removePath.Length)
```

Comment:
Hmm, what character should be used to indicate the rooted in the Git repo? I could see some folks wanting to use `$/` as opposed to `-/`. Or maybe `./` or even an emoji.  I wind up replacing `~` with 🏠.  I might be tempted to use 🎪 or 💩for the root of Git repos.  :-)

Reply:
Fair enough, `-/` is no convention.

OMG apparently `:/` is the git convention for the repository root! https://stackoverflow.com/a/22049939/27219

Additionally would you suggest adding a new configuration option like `$GitPromptSettings.DefaultPromptGitRootPrefix`?

Reply:
Cool.  Let's go with that then.  How about we call it `DefaultPromptGitDirectoryRootPrefix` to try to tie it a bit closer to the other setting?

Reply:
@rkeithhill having tried this `:/` I'm hesitating a bit.

- `:/` is supposed to be the root,
- `:/myproject/src` is incorrect based on this convention
- maybe `myproject:/src` or `[myproject]:/src` would be more correct?

Using a function (passing project name and sub-path) would allow to format that the way people want.

Reply:
<img width="319" alt="image" src="https://user-images.githubusercontent.com/297963/70249310-1e498000-1774-11ea-9bcf-8d50761a0db5.png">

Reply:
That seems reasonable. Of course, whether it's `reponame:\` or `reponame:/` will depend on the platform.  

> Using a function (passing project name and sub-path) would allow to format that the way people want.

What do you mean?  Keep in mind folks can already do this:
```
$GitPromptSettings.DefaultPromptPath.Text = '$(CallMyCustomPromptFunction)'
```

Reply:
Good point about the directory separator direction!
And you're right - one may as well just replace the function.

All fixed, but I didn't add any option for the `:` as it sounds like we are using a correct convention